### PR TITLE
RavenDB-17282 - fix NRE when using a custom converter

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17282

### Additional description

Fix NRE when using custom converter

_Please delete below the options that are not relevant_

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### CODE REVIEW

File: src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonWriter.cs [hunk 1]
```diff
@@ -359,7 +359,10 @@ public override void WriteNull()
 
         public override void WriteValue(string value)
         {
-            _manualBlittableJsonDocumentBuilder.WriteValue(value);
+            if (value != null)
+                _manualBlittableJsonDocumentBuilder.WriteValue(value);
+            else
+                _manualBlittableJsonDocumentBuilder.WriteValueNull();
```
Comment (ayende) [inline src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonWriter.cs:365]: what about6 all the other methods here that can get a null?

Comment (grisha-kotler) [reply src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonWriter.cs:365]: All other types are already handled

Comment (ayende) [reply src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonWriter.cs:365]: `LazyNumberValue` is a class, `LazyNumberValue`, the same, etc.

Comment (grisha-kotler) [reply src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonWriter.cs:365]: We can add this but it's only used for a map-reduce index where we know the type.

Comment (ppekrol) [reply src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonWriter.cs:365]: Let's add this

Comment (grisha-kotler) [reply src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonWriter.cs:365]: done

**Review by ppekrol — APPROVED** [review]

**Pull request merged**